src/parser.ts

```diff
@@ -1870,6 +1871,39 @@ export class Parser extends DiagnosticEmitter {
     // implemented methods are virtual
     if (isInterface) flags |= CommonFlags.VIRTUAL;
 
+    var declareStart = 0;
+    var declareEnd = 0;
+    var contextIsAmbient = parent.is(CommonFlags.AMBIENT);
+    if (tn.peek() == Token.DECLARE) {
+      let state = tn.mark();
+      tn.next();
+      if (tn.peek() != Token.COLON) { // modifier
```

Comment:
Seems the same applies to `public`, `protected` and `private` as well? At least the following is allowed in JS: `class B { public = 1; protected = 2; private = 3; declare = 4; }`. Perhaps there could be a more general mechanism for this?

Reply:
Also appears that not only `:` terminates here, but `(` as well, say `class B { declare() { } }`?

Reply:
Suggestion: `parseClassMemberModifier` returning the `Token` upon success or otherwise a sentinel, setting `this.modifierStart` and `this.modifierEnd` that then can be obtained?

Reply:
Is that really in this PR's scope though?

Reply:
I think the other alternative would be to not perform the check to be in line with public etc. here, and do it in a separate PR after. Would you prefer that? :) Otherwise we have inconsistency I think.

Reply:
If you're fine with it I'll do it, but it seems more fitting to do this in a separate PR